src/AstEditor.tsx

```diff
@@ -1082,3 +1098,96 @@ function FunctionEditor({
     />
   );
 }
+
+function MathEditor(props: NodeEditorProps<BinaryNode>): JSX.Element {
+  const { theme, defaultProvider } = Context.useContainer();
+  const [text, setText] = useState(serializer(props.ast));
+  const [parsing, setParsing] = useState<ParsingState>({
+    inProgress: false,
+  });
+  const changeType = () => { props.onChange(nextAst(props.ast, defaultProvider)) };
+  const parts = flattenMathParts(props.ast, props.onChange);
+
+  function onChangeText(newText: string) {
+    let error: string | undefined = undefined;
+    setParsing({
+      inProgress: true,
+      error
+    });
+    try {
+      const newAst = jsonata(newText).ast() as AST;
+      if (!isMath(newAst)) {
+        throw new Error("that's not a math expressions");
+      }
+      props.onChange(newAst);
+    } catch (e) {
+      error = "Parsing Error: " + e.message;
+    } finally {
+     setParsing({
+       inProgress: false,
+       error
+     });
+     setText(newText);
+    }
+  }
+
+  return (
+    <theme.MathEditor 
+      text={text} 
+      children={parts}
+      textChange={onChangeText} 
+      parsing={parsing} 
+      changeType={changeType}
+      {...props}
+    />
+  );
+}
+
+function flattenMathParts(ast: AST, onChange: OnChange, collectedParts: MathPart[] = []): MathPart[] {
+  function onChangeOperator(newOperator: string) {
+    if (Object.keys(Consts.mathOperators).includes(newOperator)) {
+      onChange({ ...ast, value: newOperator } as BinaryNode);
+    } else {
+      throw new Error("Not a valid math operator");
+    }
+  }
+
+  if (ast.type == "binary") {
+    flattenMathParts(
+      ast.lhs, 
+      (newAst: AST) => onChange({ ...ast, lhs: newAst }), 
+      collectedParts
+    );
+    collectedParts.push({
+     type: "operator",
+     operator: ast.value,
+     onChangeOperator
+    });
+    flattenMathParts(
+      ast.rhs, 
+      (newAst: AST) => onChange({ ...ast, rhs: newAst }), 
+      collectedParts
+    );
+  } else if (ast.type === "block") {
+     const blockParts = []
+     for (let expression of ast.expressions) {
+       flattenMathParts(expression, onChange, blockParts);
+     }
+     collectedParts.push({
+      type: "ast",
+      ast,
+      onChange,
+      children: blockParts,
+      editor: <BlockEditor ast={ast} onChange={onChange}/>
+     });
```

Comment:
I think that this is safe to remove, this is handled via sub-themes now

```suggestion
```

Reply:
Fixed